tests/test_xopen.py

```diff
@@ -223,20 +307,25 @@ def test_readline_text(fname):
         assert f.readline() == CONTENT_LINES[0]
 
 
-def test_gzip_reader_readline(gzip_reader):
+def test_reader_readline(reader):
```

Comment:
I like this approach much better: You changed a function that just tested gzip files to now also test bzip2 files. Can you go through the other places where you duplicated code and see whether you can use them same idea?